pkg/proxmox/goproxmox/ignition.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2023-2024 IONOS Cloud.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package goproxmox
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/diskfs/go-diskfs/filesystem/iso9660"
+	"github.com/luthermonson/go-proxmox"
+)
+
+var (
+	blockSize        = 2048
+	volumeIdentifier = "config-2"
+)
+
+// Ignition takes a json doc as a string and make an ISO, upload it to the data store as <vmid>-user-data.iso and will
+// mount it as a CD-ROM to be used with nutanix config-drive  'config-2'.
+func (c *APIClient) Ignition(ctx context.Context, v *proxmox.VirtualMachine, device, userdata string) error {
+	UserDataISOFormat := "user-data-%d.iso"
+	isoName := fmt.Sprintf(UserDataISOFormat, v.VMID)
+	// create userdata iso file on the local fs
+	iso, err := makeCloudInitISO(isoName, userdata)
+	if err != nil {
+		return err
+	}
+
+	// TODO, defer remove the temp file
```

Comment:
What about the TODO?

Reply:
not needed, the file will be autodeleted